rfc/0002-subdocapi.md

```diff
@@ -112,171 +89,61 @@ support empty paths. All other forms of path verification is done on the server
+### Bucket Methods
+`Bucket.LookupIn(doc_id) -> LookupBuilder`
+Creates a `LookupInBuilder` object which you can then use method-chaining to populate with lookup operations and then later execute.
 
-**FIXME:** This should really be `insert_in`, but we are already using that for
-the object-based insert.
+`Bucket.MutateIn(doc_id, cas, expiry) -> MutateBuilder`
+Creates a `MutateInBuilder` object which you can then use method-chaining to populate with mutation operations and then later execute.
 
-### `addunique_in(docid, path, value, create_parents, cas, ttl, persist_to, replicate_to)`
-Adds a 'unique' element into an array (at which end is not defined or specified).
-Functions exactly like `extend_in`. There are some server-side constraints into
-which values can be used for unique operations - currently it is restricted to
-JSON primitives (so no compound items).
+### LookupInBuilder Class
+This builder exposes the creation of a set of lookup operations to be performed.  Note that empty paths passed to either of the following methods must be considered a programming error.
 
-* Throws `PathExists` if the unique element already exists
-* Throws `CantInsertValue` if the value is not a JSON primitive.
-* Throws `PathMismatch` if the array already contains non-primitive elements
+`LookupInBuilder.Get(path) -> LookupInBuilder`
+`LookupInBuilder.Exists(path) -> LookupInBuilder`
 
-> Because there is too much variation in semantics between `add_unique_in` and
-> `extend_in`, they are separate methods rather than a single unified one.
+`LookupInBuilder.Do() -> DocumentFragment`
 
-### `counter_in(docid, path, delta, create_parents, cas, ttl, persist_to, replicate_to)`
-Performs an atomic counter operation on a given path.
+### MutateInBuilder Class
+This builder exposes the creation of a set of mutation operations to be performed.  Note that empty paths passed to the `Insert`, `Upsert`, `Replace`, `Remove`, `ArrayInsert` or `Counter` methods must be considered a programming error.
 
-* `delta`: The numeric delta, this is simply added (`+`) to the existing value.
-  This value may be signed to 'decrement' or unsigned to 'increment' the existing
-  value.
+`MutateInBuilder.Insert(path, value, createParents) -> MutateInBuilder`
+`MutateInBuilder.Upsert(path, value, createParents) -> MutateInBuilder`
+`MutateInBuilder.Replace(path, value) -> MutateInBuilder`
+`MutateInBuilder.Remove(path) -> MutateInBuilder`
+`MutateInBuilder.PushFront(path, value, createParents) -> MutateInBuilder`
+`MutateInBuilder.PushBack(path, value, createParents) -> MutateInBuilder`
+`MutateInBuilder.ArrayInsert(path, value) -> MutateInBuilder`
+`MutateInBuilder.AddUnique(path, value, createParents) -> MutateInBuilder`
+`MutateInBuilder.Counter(path, delta, createParents) -> MutateInBuilder`
 
-See "Numeric Errors" below for a list of errors thrown by this operation.
-This method also returns the current value as a `DocumentFragment`
+`MutateInBuilder.Do() -> DocumentFragment`
```

Comment:
It took me a minute understand the purpose of this method was...while the name is not terrible, i think we could do better...i am just not sure what: Execute(...), Send(), Fire(), GetResults(), Dispatch(), Run() etc....Do().

I can live with it, though...